helpers/helpers.go

```diff
@@ -343,34 +344,54 @@ func GetPublicKey() (string, error) {
 }
 
 // CheckTokenExpiration is used to determine whether the token is expiring in less than a day
-func CheckTokenExpiration(accessToken string) (bool, error) {
+func CheckTokenExpiration(accessToken string) error {
 
 	// Parse jwt token with unverifies, since we don't need to check the signatures here
 	token, _, err := new(jwt.Parser).ParseUnverified(accessToken, jwt.MapClaims{})
 	if err != nil {
-		return false, fmt.Errorf("could not parse token, reason: %s", err)
+		return fmt.Errorf("could not parse token, reason: %s", err)
 	}
 
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return fmt.Errorf("broken token (claims are empty): %v\nerror: %s", claims, err)
+	}
+
+	// Check if the token has exp claim
+	if claims["exp"] == nil {
+		return fmt.Errorf("could not parse token, reason: no expiration date")
+	}
+
+	// Parse the expiration date from token, handle cases where
+	//  the date format is nonstandard, e.g. test tokens are used
 	var expiration time.Time
-	if claims, ok := token.Claims.(jwt.MapClaims); ok {
-		// Check if the token has exp claim
-		if claims["exp"] == nil {
-			return false, fmt.Errorf("could not parse token, reason: no expiration date")
-		}
-		switch iat := claims["exp"].(type) {
-		case float64:
-			expiration = time.Unix(int64(iat), 0)
-		case json.Number:
-			tmp, _ := iat.Int64()
-			expiration = time.Unix(tmp, 0)
+	switch iat := claims["exp"].(type) {
+	case float64:
+		expiration = time.Unix(int64(iat), 0)
+	case json.Number:
+		tmp, _ := iat.Int64()
+		expiration = time.Unix(tmp, 0)
+	case string:
+		i, err := strconv.ParseInt(iat, 10, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse token, reason: %s", err)
```

Comment:
This should not really be needed.
The `"github.com/lestrrat-go/jwx/v2/jwt"` library will do this validation for you.

Reply:
Yeah, this looked a bit weird not to be handled by the library. I'll take a look at `jwx` library.